core/client/src/main/java/alluxio/client/file/options/SetAttributeOptions.java

```diff
@@ -233,7 +254,11 @@ public SetAttributeTOptions toThrift() {
     }
     if (mTtl != null) {
       options.setTtl(mTtl);
+      options.setTtlExpiryAction(mTtlExpiryAction == TtlExpiryAction.FREE
```

Comment:
this should use a `switch` with the default case setting the action to unknown so that adding a new action to TtlExpiryAction enum is not accidentally incorrectly treated by Thrift as Delete